### PR TITLE
[MU4] Added navigation for instrument settings popups

### PR DESCRIPTION
**Pull request by igorkorsukov**
I had to work hard to make everything work together - shortcuts, popup active focus, text editing, and mouse events.

Tried different solutions, the path was not winding ...
The current solution is the most stable and works well.

Navigation through the combobox has not yet been fully completed (I will do separately)

https://user-images.githubusercontent.com/3818029/115593986-80594580-a2d5-11eb-8295-ac93dadbcdc9.mp4

### CODE REVIEW

File: src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml [hunk 1]
```diff
@@ -127,29 +170,29 @@ ComboBox {
                     when: index === 0
 
                     PropertyChanges { target: delegateBackgroundRect; topLeftRadius: 4
-                                                                      topRightRadius: 4
-                                                                      bottomLeftRadius: 0
-                                                                      bottomRightRadius: 0 }
+                        topRightRadius: 4
+                        bottomLeftRadius: 0
+                        bottomRightRadius: 0 }
```
Comment (igorkorsukov) [inline src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml:175]: auto indentation (Ctrl+A Ctrl+I)

File: src/userscores/qml/MuseScore/UserScores/internal/TimeSignatureSettings.qml [hunk 2]
```diff
@@ -38,8 +38,8 @@ FlatButton {
     TimeSignatureView {
         id: timeSignatureView
 
-        anchors.horizontalCenter: root.horizontalCenter
-        anchors.verticalCenter: root.verticalCenter
+        anchors.horizontalCenter: parent.horizontalCenter
```
Comment (vpereverzev) [inline src/userscores/qml/MuseScore/UserScores/internal/TimeSignatureSettings.qml:41]: makes sense to use anchors.centeredIn: parent

File: src/framework/uicomponents/view/popupview.h [hunk 3]
```diff
@@ -24,13 +24,13 @@
...
 
 #include <QQuickItem>
-#include <QQuickView>
 #include <QQmlParserStatus>
 
 #include "modularity/ioc.h"
 #include "ui/imainwindow.h"
 
 namespace mu::uicomponents {
+class MyQuickView;
```
Comment (vpereverzev) [inline src/framework/uicomponents/view/popupview.h:33]: Let's rename it into "QuickViewWrapper or Adapter or Handler"

File: src/framework/uicomponents/view/popupview.cpp [hunk 4]
```diff
@@ -30,13 +31,142 @@
 
 #include "log.h"
 
+namespace mu::uicomponents {
```
Comment (vpereverzev) [inline src/framework/uicomponents/view/popupview.cpp:34]: I'd suggest to move this class into ../internal

File: src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml [hunk 5]
```diff
@@ -216,10 +216,8 @@ Rectangle {
         hoverEnabled: true
 
         onPressed: {
-            if (!valueInput.activeFocus) {
-                valueInput.forceActiveFocus()
-            }
-
+            root.insureActiveFocus()
```
Comment (vpereverzev) [inline src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml:219]: might be better to say "ensure"

File: src/userscores/qml/MuseScore/UserScores/internal/TimeSignatureSettings.qml (hunk 2 above)
Comment (igorkorsukov) [reply src/userscores/qml/MuseScore/UserScores/internal/TimeSignatureSettings.qml:41]: Maybe, I just fixed, what I broken

File: src/framework/uicomponents/view/popupview.h (hunk 3 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/view/popupview.h:33]: Do we need to complicate?
I usually use the prefix `my` if the scope is only for a given object.

File: src/framework/uicomponents/view/popupview.cpp (hunk 4 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/view/popupview.cpp:34]: I'm not sure, it does not have independent value, it's just part of the implementation of `PopupView`

File: src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml (hunk 1 above)
Comment (shoogle) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml:175]: Is there a reason for having `PropertyChanges { target: delegateBackgroundRect; topLeftRadius: 4` all on the same line, or could it be:

```qml
PropertyChanges {
    target: delegateBackgroundRect;
    topLeftRadius: 4
```

Also, could the closing `}` go on a new line after `bottomRightRadius: 0`?

File: src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml (hunk 5 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml:219]: I will rename it in a next PR

**Review by shoogle — COMMENTED** [review]
When using MuseScore in fullscreen mode on macOS, for some reason this PR causes the Instrument Settings popup to open in a new virtual desktop rather than on the same screen as the application. This wasn't happening before this PR.

**Comment by igorkorsukov** [comment]
> When using MuseScore in fullscreen mode on macOS, for some reason this PR causes the Instrument Settings popup to open in a new virtual desktop rather than on the same screen as the application. This wasn't happening before this PR.

Thank you, so we still need to work on the pop-up view.

**Comment by igorkorsukov** [comment]
> When using MuseScore in fullscreen mode on macOS, for some reason this PR causes the Instrument Settings popup to open in a new virtual desktop rather than on the same screen as the application. This wasn't happening before this PR.

Unfortunately, everything works as expected for me, it makes things a little bit difficult :)

https://user-images.githubusercontent.com/3818029/115710145-50ac4b00-a372-11eb-9897-264cdc62c2ad.mp4

File: src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml (hunk 1 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/StyledComboBox.qml]: reformatted the code

File: src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml (hunk 5 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml]: renamed now

File: src/framework/uicomponents/view/popupview.h (hunk 3 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/view/popupview.h]: Renamed and restructured

**Review by vpereverzev — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
So the problem is apparently macOS only...
For me, the popup appears on a new virtual desktop, produces some graphical glitches and makes MuseScore completely unresponsive, so the only option is to force-quit it. 

https://user-images.githubusercontent.com/48658420/115849461-31bac100-a425-11eb-9300-62d730d5851b.mov

**Comment by igorkorsukov** [comment]
> So the problem is apparently macOS only...
> For me, the popup appears on a new virtual desktop, produces some graphical glitches and makes MuseScore completely unresponsive, so the only option is to force-quit it.
> 
>  Schermopname.2021-04-23.om.11.10.59.mov

Thanks, I'm going to investigate and fix this shortly.

**Pull request merged**